'''
Подвиг 7. В практике программирования блок else используют как элемент отладки программы: в него прописывают текст программы, в котором заведомо не произойдет исключений, отлавливаемых в блоке try. Выполним на практике такой пример.

Вам необходимо объявить функцию с сигнатурой:

def get_loss(w1, w2, w3, w4): ...

где w1, w2, w3, w4 - любые числа. Функция должна возвращать значение, вычисленное по формуле:

y = 10 * w1 // w2 - 5 * w2 * w3 + w4

Здесь фрагмент вычисления w1 // w2 содержит потенциальную ошибку деления на ноль, поэтому его следует делать в блоке try. А в блоке else продолжить вычисления, где не используются операции деления.

Если происходит деление на ноль, то функция должна возвращать строку:

"деление на ноль"

P.S. В программе нужно объявить только функцию. Вызывать ее и выводить на экран ничего не нужно.Memory limit: 256 MBTime limit: 15 seconds

'''


def get_loss(w1, w2, w3, w4):
    '''Функция возвращающая вычисления или сообщения об ошибке деления на ноль.'''

    mul = 10 * w1
    try:
        div = mul // w2
    except ZeroDivisionError:
        return 'деление на ноль'
    res = div - 5 * w2 * w3 + w4
    return res


# Для проверки
print(get_loss(2, 2, 4, 5))
print(get_loss(2, 0, 4, 5))
